r55/src/main.rs

```diff
@@ -187,7 +188,230 @@ fn test_deploy() {
+    let selector_transfer: u32 = 1;
+    
+    let address1 = address!("0000000000000000000000000000000000000001");
+    let address2 = address!("0000000000000000000000000000000000000002");
+    
+    // Mint tokens
+    let mint_value: u64 = 5000;
+    let mut calldata_mint = (address1, mint_value).abi_encode();
+    let mut complete_calldata_mint = selector_mint.to_le_bytes().to_vec();
+    complete_calldata_mint.append(&mut calldata_mint);
+    run_tx(&mut db, &CONTRACT_ADDR, complete_calldata_mint);
+
+    // Transfer tokens
+    let transfer_value: u64 = 750;
+    let mut calldata_transfer = (address1, address2, transfer_value).abi_encode();
+    let mut complete_calldata_transfer = selector_transfer.to_le_bytes().to_vec();
+    complete_calldata_transfer.append(&mut calldata_transfer);
+    
+    let transfer_result = run_tx(&mut db, &CONTRACT_ADDR, complete_calldata_transfer);
+
+    assert_eq!(transfer_result.logs.len(), 1, "Should have exactly 1 log");
+    
+    analyze_transfer_log(
+        &transfer_result.logs[0],
+        address1,
+        address2,
+        transfer_value
+    );
+}
+
+
+fn test_transfer_logs_simple() {
+    let rv_bytecode = compile_runtime("erc20").unwrap();
+    const CONTRACT_ADDR: Address = address!("0d4a11d5EEaaC28EC3F61d100daF4d40471f1852");
+    let mut db = InMemoryDB::default();
+
+    let mut bytecode = vec![0xff];
+    bytecode.extend_from_slice(&rv_bytecode);
+    let bytecode = Bytes::from(bytecode);
+    add_contract_to_db(&mut db, CONTRACT_ADDR, bytecode);
+
+    let address1 = address!("0000000000000000000000000000000000000001");
+    let address2 = address!("0000000000000000000000000000000000000002");
+    
+    // Mint
+    let selector_mint: u32 = 2;
+    let mut calldata_mint = (address1, 5000u64).abi_encode();
+    let mut complete_calldata_mint = selector_mint.to_le_bytes().to_vec();
+    complete_calldata_mint.append(&mut calldata_mint);
+    run_tx(&mut db, &CONTRACT_ADDR, complete_calldata_mint);
+
+    // Transfer
+    let selector_transfer: u32 = 1;
+    let mut calldata_transfer = (address1, address2, 750u64).abi_encode();
+    let mut complete_calldata_transfer = selector_transfer.to_le_bytes().to_vec();
+    complete_calldata_transfer.append(&mut calldata_transfer);
+    
+    let transfer_result = run_tx(&mut db, &CONTRACT_ADDR, complete_calldata_transfer);
+
+    println!("\n=== Transfer Log Output ===");
+    for log in transfer_result.logs.iter() {
+        let data = log.data.data.as_ref();
+        
+        let from_hex = hex::encode(&data[..20]);
+        let to_hex = hex::encode(&data[20..40]);
+        let value = u64::from_ne_bytes(data[40..48].try_into().unwrap());
+
+        println!("From: 0x{}", from_hex);
+        println!("To: 0x{}", to_hex);
+        println!("Value: {}", value);
+    }
+}
 fn main() {
     test_runtime_from_binary();
     test_deploy();
+    test_erc20_mint_and_transfer();
+    test_transfer_with_logs();
+    test_transfer_logs_simple();
```

Comment:
Nice!

We should probably move these into a separate testing infra at some point.